File: app/src/main/java/com/huck/biblequiz/data/remote/BollsApiService.kt
package com.huck.biblequiz.data.remote

import com.huck.biblequiz.data.remote.dto.BookDto
import com.huck.biblequiz.data.remote.dto.VerseDto
import retrofit2.http.GET
import retrofit2.http.Path

interface BollsApiService {

    @GET("get-books/NKJV/")
    suspend fun getBooks(): List<BookDto>

    @GET("get-text/NKJV/{bookId}/{chapter}/")
    suspend fun getChapter(
        @Path("bookId") bookId: Int,
        @Path("chapter") chapter: Int
    ): List<VerseDto>
}

File: ios/BeSpecific/Data/Remote/BollsAPIService.swift
import Foundation

actor BollsAPIService {
    private let baseURL = "https://bolls.life/"
    private let session: URLSession

    init(session: URLSession = .shared) {
        self.session = session
    }

    func getBooks() async throws -> [BookDTO] {
        let url = URL(string: "\(baseURL)get-books/NKJV/")!
        let (data, _) = try await session.data(from: url)
        return try JSONDecoder().decode([BookDTO].self, from: data)
    }

    func getChapter(bookId: Int, chapter: Int) async throws -> [VerseDTO] {
        let url = URL(string: "\(baseURL)get-text/NKJV/\(bookId)/\(chapter)/")!
        let (data, _) = try await session.data(from: url)
        return try JSONDecoder().decode([VerseDTO].self, from: data)
    }
}
